File: mobile/swish/lib/models/video.dart
import 'channel.dart';

class Content {
  final String assetID;
  final String playbackID;

  Content.fromJSON(Map<String, dynamic> json)
      : assetID = json['assetID'],
        playbackID = json['playbackID'];
}

class Video {
  final String id;
  final String author;
  final String title;
  final String description;
  final Channel channel;
  final Content content;
  final int uploadDate;

  Video.fromJSON(Map<String, dynamic> json)
      : id = json['id'],
        author = json['author'],
        title = json['title'],
        description = json['description'],
        uploadDate = json['uploadDate'],
        channel = Channel.fromJSON(json['channel']),
        content = Content.fromJSON(json['content']);
}

File: web/swish/models/video.ts
import { IChannel } from './channel';

export interface IVideo {
  id: string;
  title: string;
  author: string;
  description: string;
  uploadDate: number;
  channel: IChannel;
  content?: IVideoContent;
}

export interface IVideoContent {
  id: string;
  assetID: string;
  playbackID: string;
  duration: number;
}

export interface IVideoUpdate {
  title?: string;
  description?: string;
}

export interface IVideoSearchObject {
  objectID: string;
  type: 'video';
  title: string;
  description: string;
  uploadDate: number;
}

export interface IVideoFragment {
  t1: number;
  t2: number;
}

export interface IVideoSegment {
  index: number;
}
